prime-router/metadata/hl7_mapping/common/datatype/ce-coded-with-exceptions.yml

```
@@ -1,47 +1,156 @@
 elements:
   - name: identifier-code
-    value: [ '%resource.coding[0].code' ]
+    value: [ '%resource.coding.where(extension(%`rsext-code-index-name`).value = "identifier").code' ]
```

You should be able to combine the two cases into one element with something like
```suggestion
    value: [ '%resource.coding.where(extension(%` rsext-code-index-name` ).value = "identifier").code', '%resource.coding[0].code' ]
```
You might need to inspect the code or test to make sure that ordering is correct, but if one expression evaluates to null, it'll try the next in the list until it finds one that works.

---

Interesting... let me play around with this.  There are a few places where I don't think that will work since the condition needs to look at the resource (specifically system), but maybe can clean up the others

---

Tried a couple of things, but I'm leaning towards leaving it the same as this can't be done for every case because there are situations where the coding array is `['identifier', 'second alternate identifier']`

---

Talked on slack about adding this for the primary identifier too in case we end up with an array like `["alternate identifier", "second alternate identifier"]`, Michael is planning on adding it in just to be safe.